src/server/conn_context.cc

```diff
@@ -230,6 +237,13 @@ void ConnectionContext::OnClose() {
     token.Wait();
     DCHECK(!conn_state.subscribe_info);
   }
+
+  if (!conn_state.exec_info.watched_keys.empty()) {
+    shard_set->RunBriefInParallel([this](EngineShard* shard){
+      // TODO: filter keys in shard?
+      return shard->db_slice().UnregisterWatches(&conn_state.exec_info);
```

Comment:
I guess there is actually no point in filtering them, because a `watched_keys_` access probably takes the same time

Reply:
yes, it's premature optimization.